lib/web/desktop.go

```diff
@@ -495,28 +495,15 @@ func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn) error {
 	go func() {
 		defer closeOnce.Do(close)
 
-		buf := make([]byte, 4096)
-		for {
-			_, reader, err := ws.NextReader()
-			switch {
-			case utils.IsOKNetworkError(err):
-				errs <- nil
-				return
-			case err != nil:
-				errs <- err
-				return
-			}
-			n, err := reader.Read(buf)
-			if err != nil {
-				errs <- err
-				return
-			}
-
-			if _, err := wds.Write(buf[:n]); err != nil {
-				errs <- trace.Wrap(err, "sending TDP message to desktop agent")
-				return
-			}
+		// io.Copy is fine here, as the Windows Desktop Service
+		// operates on a stream and doesn't care if TPD messages
+		// are fragmented
+		stream := &WebsocketIO{Conn: ws}
+		_, err := io.Copy(wds, stream)
+		if utils.IsOKNetworkError(err) {
+			err = nil
 		}
+		errs <- err
```

Comment:
Here's how I would write it. This keeps the behavior of reusing the same read buf for each message, but uses a `bytes.Buffer` that can grow rather than a fixed-capacity slice. (While still making it easy to check the first byte on v15)
```suggestion
		var buf bytes.Buffer
		for {
			_, reader, err := ws.NextReader()
			switch {
			case utils.IsOKNetworkError(err):
				errs <- nil
				return
			case err != nil:
				errs <- err
				return
			}
			buf.Reset()
			if _, err := io.Copy(&buf, reader); err != nil {
				errs <- err
				return
			}

			if _, err := wds.Write(buf.Bytes()); err != nil {
				errs <- trace.Wrap(err, "sending TDP message to desktop agent")
				return
			}
		}
```